components/clientComponents/forms/TextInput/TextInput.tsx

```diff
@@ -59,17 +59,48 @@ export const TextInput = (
   return (
     <>
       {meta.error && <ErrorMessage id={"errorMessage" + id}>{meta.error}</ErrorMessage>}
+      {/* For AT to help identify the input as a number */}
+      {type === "number" && (
+        <div id={`${id}-description-number`} hidden>
+          {t("number")}
+        </div>
+      )}
       <input
         data-testid="textInput"
         className={classes}
         id={id}
-        type={type}
+        type={type === "number" ? "text" : type}
         required={required}
         autoComplete={autoComplete ? autoComplete : "off"}
         placeholder={placeholder}
         {...ariaDescribedByIds()}
         {...field}
         onChange={handleTextInputChange}
+        // Note: not using type=number for numbers for UX reasons.
+        // See: #4851 and https://tinyurl.com/2p9tm5vk
+        {...(type === "number" && {
+          // For mobile phones to switch the keypad to numeric
+          inputMode: "numeric",
+          "aria-describedby": `${id}-description-number`,
+          // Note: "onBeforeInput" and e.data could also be used but I'm not sure how cross-browser
+          // consistent it is
+          onKeyDown: (e) => {
+            // Allow control keys
+            if (
+              e.key.includes("Arrow") ||
```

Comment:
Did you find any posts about a11y + using arrow keys with the number input ?

With a "browser" / html number input I can press and hold arrow up or down to increment decrement numbers.  Does removing this functionality cause issues ?

Reply:
Actually that was one of the UX issues with the input=number. Because the stepper is controlled by the mouse, if the input is focussed a user can easily accidentally scroll and increase/decrease the number without meaning to. 

The type=number is accessible and meets WCAG. It's just the UX issues e.g. https://tinyurl.com/2p9tm5vk

The people I respect in the accessibility community seem to be pretty neutral on this.

Though MDN recommends text unless you have a specific reason for the spin button integer entry:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#:~:text=as%20explained%20above).-,Accessibility,to%20numbers%20and%20associated%20characters.